src/main/java/com/streamr/client/dataunion/DataUnion.java

```diff
@@ -0,0 +1,231 @@
+    private final DataUnionSidechain sidechain;
+    private final Web3j mainnetConnector, sidechainConnector;
+    private final Credentials mainnetCred, sidechainCred;
+
+    //use DataUnionClient to instantiate
+    protected DataUnion(DataUnionMainnet mainnet, Web3j mainnetConnector, Credentials mainnetCred, DataUnionSidechain sidechain, Web3j sidechainConnector, Credentials sidechainCred) {
+        this.mainnet = mainnet;
+        this.mainnetConnector = mainnetConnector;
+        this.mainnetCred = mainnetCred;
+        this.sidechain = sidechain;
+        this.sidechainConnector = sidechainConnector;
+        this.sidechainCred = sidechainCred;
+    }
+
+    public boolean waitForDeployment(long pollInterval, long timeout) throws Exception {
+        String code = waitForCodeAtAddress(sidechain.getContractAddress(), sidechainConnector, pollInterval, timeout);
+        return code != null && !code.equals("0x");
+    }
+
+    public String getMainnetContractAddress(){
+        return mainnet.getContractAddress();
+    }
+
+    public String getSidechainContractAddress(){
+        return sidechain.getContractAddress();
+    }
+
+    public boolean isDeployed() throws Exception {
+        //this will check the condition only once:
+        return waitForDeployment(0,0);
+    }
+
+    public void sendTokensToBridge() throws Exception {
+        mainnet.sendTokensToBridge().send();
+    }
+
+    public BigInteger waitForEarningsChange(final BigInteger initialBalance, long pollInterval, long timeout) throws Exception {
+        Web3jUtils.Condition earningsChange = new Web3jUtils.Condition(){
+            @Override
+            public Object check() throws Exception {
+                BigInteger bal = sidechain.totalEarnings().send().getValue();
+                if(!bal.equals(initialBalance)) {
+                    return bal;
+                }
+                return null;
+            }
+        };
+        return (BigInteger) waitForCondition(earningsChange, pollInterval, timeout);
+    }
+
+    public EthereumTransactionReceipt addJoinPartAgents(String ... agents) throws Exception {
+        return new EthereumTransactionReceipt(sidechain.addJoinPartAgents(asDynamicAddressArray(agents)).send());
+    }
+
+    public EthereumTransactionReceipt partMembers(String ... members) throws Exception {
+        return new EthereumTransactionReceipt(sidechain.partMembers(asDynamicAddressArray(members)).send());
+    }
+
+    public EthereumTransactionReceipt addMembers(String ... members) throws Exception {
+        return new EthereumTransactionReceipt(sidechain.addMembers(asDynamicAddressArray(members)).send());
+    }
+
+
+    /**
+     *
+     * withdraw a member to his own address. Must be that member or admin.
+
+     * @param member
+     * @param amount amount in wei
+     * @return
+     * @throws Exception
+     */
+    public EthereumTransactionReceipt withdrawAllTokensForSelfOrAsAdmin(String member, BigInteger amount) throws Exception {
```

Comment:
Copy-paste error: should be `withdrawTokensForSelfOrAsAdmin`

Reply:
y I'll fix